src/Extension/ExtensionManager.php

```diff
@@ -86,7 +86,9 @@ public function getExtensions()
             // We calculate and store a set of composer package names for all installed Flarum extensions,
             // so we know what is and isn't a flarum extension in `calculateDependencies`.
             // Using keys of an associative array allows us to do these checks in constant time.
+            // We do the same for enabled extensions, for optional dependencies.
             $installedSet = [];
+            $enabledIds = array_flip($this->getEnabled());
```

Comment:
This is probably just a nitpick, but leaving it unflipped and then using `in_array` instead of `array_key_exists` seems more simple

Reply:
The rationale here was to use constant time key-based lookup instead of linear time value, since multiple extensions will perform multiple lookups.

Reply:
I see, I hadn't noticed or thought about it before, that's better performance wise, and since we're certain that the values are unique than it works just fine, nice.

Reply:
Any reason for not explicitly approving the PR?

Reply:
running some tests locally right now